clients/src/main/java/org/apache/kafka/common/requests/BrokerRegistrationRequest.java

```diff
@@ -45,6 +45,10 @@ public short oldestAllowedVersion() {
 
         @Override
         public BrokerRegistrationRequest build(short version) {
+            if (version < 2) {
+                // Reset the PreviousBrokerEpoch to default if not supported.
+                data.setPreviousBrokerEpoch(new BrokerRegistrationRequestData().previousBrokerEpoch());
```

Comment:
This stuck me as a little backwards. Usually in request builders we will only set a field if it's supported at the version, e.g.

```
if (version >= 2) data.setPreviousBrokerEpoch
```

but I see we're not really using the builder pattern in BrokerRegistrationRequest. The caller is just creating the BrokerRegistrationRequestData directly and passing it into this builer. 

This fix is okay for now, I'll file a follow-up JIRA to refactor this builder class a bit

Reply:
Thanks David!